packages/echo-core/src/types/common.ts

```diff
@@ -1,5 +1,9 @@
 export type NestedPartial<T> = {
-    [P in keyof T]?: T[P] extends Array<infer U> ? Array<Partial<U>> : T[P] extends {} ? Partial<T[P]> : T[P];
+    [P in keyof T]?: T[P] extends Array<infer U>
+        ? Array<Partial<U>>
+        : T[P] extends Record<string, never> // empty object
```

Comment:
New version of eslint complained about the {} (empty object) as type.

Got this error message:
![core-lint-empty-object](https://user-images.githubusercontent.com/13872908/144439999-1e2f5ac8-4343-4cb2-a2b8-30d58bf2caa7.png)
Updated accordingly.